docs/parser/comparator/README.md

```diff
@@ -0,0 +1,135 @@
+docker run --rm -it -p 8001:80 -v (pwd)/parser-php-php-pegjs.php:/parser.php -v (pwd)/post.peg.php:/post.peg.php php-parser
+```
+
+The defined interface for the PHP runner is that we provide a class
+called `MyParser` which has a method called `parse()`. Since we're
+using the spec parser from Gutenberg it doesn't provide that interface,
+thus we write a small wrapper file that handles that translation. This
+wrapper is mounted as `/parser.php` but since it requires the PEG-generated
+parser we also mount that at a known location: `/post.peg.php`
+
+The provided Dockerfile creates a web-server that listens to port 80. We
+will have multiple parsers running at once so we'll map that port to another
+open port - in this case port 8001.
+
+Now the Javascript.
+
+```
+# Build the provided docker image
+docker build -t js-parser -f runner-js-sync.Dockerfile .
+
+# Run it
+docker run --rm -it -p 8002:80 -v (pwd)/parser-js-pegjs.js:/parser.js -v (pwd)/post.peg.js:/post.peg.js js-parser
+```
+
+The options here should make sense as well. We created a small wrapper then
+setup another port on which to run this.
+
+When running the comparator we will use `localhost:8001` and `localhost:8002`
+as the URLs for the parsers.
+
+We should not that Docker was unnecessary. As long as we provide a web service
+conforming to the requirements we can run anything and we can skip docker.
+
+## Web interface
+
+In order to normalize runs across all languages, across synchronous vs. asynchronous
+code, across all environments, we are connecting via `POST` calls to a web server.
+We are also trusting the parser under test to report its own metrics. We're not
+looking for people to cheat - we're trying to honestly compare one parser against
+another.
+
+The request may contain a `count` parameter indicating how many times to run the
+document through the parse function. This can be helpful in order to mitigate
+the effects of caches and data locality and other non-obvious factors.
+
+The body of the request should be a raw string which is the content of the document
+to be parsed.
+
+The parser should then parse the document and track the time it took to parse it
+and do its best to measure the memory used by the parser. Memory can be tricky to
+measure; ideally we want the total amount of <abbr>rss</abbr> memory as reported
+in `/proc/{pid}` on a Linux system.
+
+The response should mirror this format in a JSON-encoded value:
+
+```
+{
+	parse: [array of blocks] // parsed output of the document
+	us: [number] // number of milliseconds taken to parse the document across all runs
+	usAvg: [number] // number of milliseconds on average it took to parse the document once
+	rss: {
+		start: [number] // bytes of used memory before any initialization
+		beforeParserInit: [number] // bytes right before initializing parser
+		afterParserInit: [number] // bytes right after initializing parser
+		end: [number] // bytes after parsing document the requested number of times
+	}
```

Comment:
What about `sentinel`?

Reply:
it's only there to try and de-optimize the loop to make sure the compiler doesn't try and do anything funny by not running code

it's not important to the output format. I considered adding it in there but I ended up leaving it irrelevant